### PR TITLE
Changed calculations of savings

### LINKED ISSUE #49: Adjust cost calculation to include fixed-costs ("checkpoints") with revocation increases as well
**Is your feature request related to a problem? Please describe.**
In the web component, a user can increase or decrease revocations (individuals removed from community supervision and sent to prison). The component calculates the amount that it will cost (or that will be saved) if revocations go up or down.

Currently, if revocations go **down** then the savings is calculated both using a 'Marginal cost per inmate' and adding in the 'associated savings' at each of several checkpoints. These checkpoints represent a facility being closed.

However, when revocations go **up** we only increase marginal costs - we do not add the fixed costs for new facilities as the population in prison increases (see attached images - note that costs increase at a much slower rate than savings).

**Describe the solution you'd like**
We'd like to use the same checkpoints and associated savings for revocations going up. That is, we use marginal costs until we hit a checkpoint, and then costs increase by that checkpoint amount - then costs continue to rise by the marginal cost per inmate until we hit the next checkpoint.

**Describe alternatives you've considered**
No alternatives have been considered. We could leave the behavior as-is, but it wouldn't accurately reflect the costs of incarceration.

fast_increase_in_savings
slow_increase_in_costs

### PR DESCRIPTION
**Pull request by Wmaileq**
## Description of the change

- Removed checkpoints from input data and replaced with number of facilities and state capacity columns
- changed calcSavings function

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #49 

### Development

These boxes should be checked by the submitter prior to merging:

- [X] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/SupervisionSuccess/model/calcSavings.js [hunk 1]
```diff
@@ -19,24 +19,34 @@ const MONTHS_IN_YEAR = 12;
...
+  const facilitiesDiffRatio = (1 - totalPopulation / stateWideCapacity) * numberOfFacilities;
+
+  const facilitiesDiff = Math.floor(facilitiesDiffRatio);
+  const restPopulation = facilitiesDiffRatio % 1;
+
+  const graduatedSavings =
+    facilitiesDiff * (stateWideCapacity / numberOfFacilities) * totalCostPerInmate;
+
+  const marginalSavings =
+    restPopulation * (stateWideCapacity / numberOfFacilities) * marginalCostPerInmate;
```
Comment (Wmaileq) [inline src/SupervisionSuccess/model/calcSavings.js:49]: It looks like here we need to multiply `restPopulation` just on `marginalCostPerInmate`, and not on `(stateWideCapacity / numberOfFacilities)`

File: src/SupervisionSuccess/model/calcSavings.js [hunk 2]
```diff
@@ -19,24 +19,34 @@ const MONTHS_IN_YEAR = 12;
...
-  };
-  const graduatedSavings = reachedCheckPoint.savings;
-  const marginalSavings = (revocationsDiff - reachedCheckPoint.checkpoint) * marginalCostPerInmate;
+  const facilitiesDiffRatio = (1 - totalPopulation / stateWideCapacity) * numberOfFacilities;
+
+  const facilitiesDiff = Math.floor(facilitiesDiffRatio);
+  const restPopulation = facilitiesDiffRatio % 1;
+
+  const graduatedSavings =
+    facilitiesDiff * (stateWideCapacity / numberOfFacilities) * totalCostPerInmate;
```
Comment (Wmaileq) [inline src/SupervisionSuccess/model/calcSavings.js:46]: Here is the difference between the excel model. Copy-pasted from the model it should look like:

`const graduatedSavings = facilitiesDiff * stateWideCapacity * totalCostPerInmate;`

But as `stateWidthCapacity` is the capacity of all state facilities (not the average capacity of a facility), I think dividing on numberOfFacilities makes sense.

**Review by jessex — COMMENTED** [review]
I'm going to hold off on an in-depth review until your questions for @alpaco42 about the updated formula have been resolved. But I did want to point out that I ran a quick test locally and the results for Texas, specifically, are quite strange. I've attached two screenshots below. Is this because of something wrong or under question about the updated formula? Or is there a bug in the Texas params somewhere, perhaps?

Screen Shot 2020-12-18 at 10 49 25

Screen Shot 2020-12-18 at 10 49 16

**Comment by alpaco42** [comment]
Left an update on the ticket for this PR, updated google sheet should address issues flagged here (@jessex should also fix the exploding costs, but let me know if it doesn't!)

**Comment by jessex** [comment]
Thanks, @alpaco42!

@Wmaileq, let me know when you've made further updates based on Paco's comments on the ticket and I'll do another review-and-test here. Thanks!

**Comment by Wmaileq** [comment]
@jessex I will add cost increase tests after my question in issue will be resolved.

**Comment by jessex** [comment]
@Wmaileq I pulled down your latest commit and tested again and I am still seeing those enormous numbers for Texas specifically. Have you tested your changes with Texas yourself? If not, please do so at your next available opportunity. Thanks!

**Comment by Wmaileq** [comment]
@jessex the Texas input data problem solved. Added tests, I think it is ready to be shipped.

**Review by jessex — APPROVED** [review]
Well done, @Wmaileq! I think we're set here.

**Pull request merged**